provider/blob/src/export/blob_exporter.rs

```diff
@@ -149,24 +157,32 @@ impl BlobExporter<'_> {
 
         let all_markers = self.all_markers.lock().expect("poison");
         let resources = self.resources.lock().expect("poison");
+        let checksums = self.checksums.lock().expect("poison");
 
         let markers: ZeroVec<DataMarkerIdHash> = all_markers.iter().copied().collect();
 
         let locales_vec: Vec<Vec<u8>> = all_markers
             .iter()
-            .map(|marker_path_hash| resources.get(marker_path_hash))
-            .map(|option_sub_map| {
-                if let Some(sub_map) = option_sub_map {
-                    let mut sub_map = sub_map.clone();
-                    sub_map
-                        .iter_mut()
-                        .for_each(|(_, id)| *id = *remap.get(id).expect("in-bound index"));
-                    let zerotrie = ZeroTrieSimpleAscii::try_from(&sub_map).expect("in-bounds");
-                    zerotrie.into_store()
-                } else {
-                    // Key with no locales: insert an empty ZeroTrie
-                    ZeroTrieSimpleAscii::default().into_store()
+            .map(|marker_path_hash| {
+                (
+                    resources.get(marker_path_hash),
+                    checksums.get(marker_path_hash),
+                )
+            })
+            .map(|(option_sub_map, checksum)| {
+                let mut sub_map = BTreeMap::new();
+                if let Some(sub_map_wrong) = option_sub_map {
+                    if let Some(&checksum) = checksum {
+                        // TODO how does this behave on 32-bit platforms?
+                        sub_map.insert(CHECKSUM_KEY, checksum as usize);
```

Comment:
The checksum should be a `u64` not a `usize` right?

Reply:
If I put a usize value into a `ZeroAsciiTrie` on a 64-bit system, and then access it on a 32-bit system, what will have happened to it? How many bits are stored on the wire.

Reply:
It uses `usize` in the API because it doesn't really care what integer is stored; it stores a varint. Most use cases of ZeroTrie have involved storing indices into an array of data sitting on the side, so that's mainly what the API was optimized for.

Reply:
We need a way to get 64 bits out regardless off the usize length then.

Reply:
I suggest adding an API to ZeroTrie to get/set unsigned ints of specific sizes.